aten/src/ATen/TensorIterator.cpp

```diff
@@ -177,6 +177,7 @@ TensorIteratorConfig& TensorIteratorConfig::declare_static_shape(IntArrayRef sha
 
 TensorIteratorConfig& TensorIteratorConfig::declare_static_shape(IntArrayRef shape, IntArrayRef squash_dims) {
   declare_static_shape(shape);
+  // NOLINTNEXTLINE(bugprone-unchecked-optional-access)
```

Comment:
Shouldn't it also just return if static_shape_ is empty too? Or do we want it to error here?

Reply:
This checker is not sound in that it can infer and remove the impossible branch. In this case it would throw in  declare_static_shape if shape is null opt.